internal/envoy/v3/cluster.go

```diff
@@ -207,7 +210,7 @@ func DNSNameCluster(c *dag.DNSNameCluster) *envoy_cluster_v3.Cluster {
 
 	var transportSocket *envoy_core_v3.TransportSocket
 	if c.Scheme == "https" {
-		transportSocket = UpstreamTLSTransportSocket(UpstreamTLSContext(c.UpstreamValidation, c.Address, nil))
+		transportSocket = UpstreamTLSTransportSocket(UpstreamTLSContext(c.UpstreamValidation, c.Address, nil, c.UpstreamTLS))
```

Comment:
Is `c.UpstreamTLS` unset? I believe it should be set here https://github.com/KauzClay/contour/blob/5233293fc71feaf8461e619d6ca12257f46c78de/internal/dag/httpproxy_processor.go#L490-L495

Reply:
thank you for pointing this out, I had a feeling some things were missing, but I got a bit lost in the code.